admin-dev/themes/new-theme/js/components/modify-all-shops-checkbox/index.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import MultiShopModifyAllMap from './component-map';
+
+const {$} = window;
+
+/**
+ * Enables multi shop modify all checkboxes on the page. The checkboxes are hidden by default,
+ * they appear on input focus and stay visible when the value changed.
+ */
+export default class ModifyAllShopsCheckbox {
```

Comment:
The name looks really too generic, I'm unable to find what this component is doing even with the description and it looks not adapted to a class, what about something like `AllShopContextCheckbox` or if you have a better idea..